src/install/lockfile.zig

```diff
@@ -2390,6 +2396,121 @@ pub const Package = extern struct {
         );
     }
 
+    fn parseDependency(
+        lockfile: *Lockfile,
+        allocator: std.mem.Allocator,
+        log: *logger.Log,
+        source: logger.Source,
+        comptime group: DependencyGroup,
+        string_builder: *StringBuilder,
+        comptime features: Features,
+        package_dependencies: []Dependency,
+        dependencies: []Dependency,
+        tag: ?Dependency.Version.Tag,
+        workspace_path: ?String,
+        external_name: ExternalString,
+        version: string,
+        key: Expr,
+        value: Expr,
+    ) !?Dependency {
+        const external_version = string_builder.append(String, version);
+        var buf = lockfile.buffers.string_bytes.items;
+        const sliced = external_version.sliced(buf);
+
+        var dependency_version = Dependency.parseWithOptionalTag(
+            allocator,
+            sliced.slice,
+            tag,
+            &sliced,
+            log,
+        ) orelse Dependency.Version{};
+
+        switch (dependency_version.tag) {
+            .folder => {
+                dependency_version.value.folder = string_builder.append(
+                    String,
+                    Path.relative(
+                        FileSystem.instance.top_level_dir,
+                        Path.joinAbsString(
+                            FileSystem.instance.top_level_dir,
+                            &[_]string{
+                                source.path.name.dir,
+                                dependency_version.value.folder.slice(buf),
+                            },
+                            .posix,
+                        ),
+                    ),
+                );
+            },
+            .workspace => if (workspace_path) |path| {
+                dependency_version.value.workspace = path;
+            } else {
+                const path = string_builder.append(
+                    String,
+                    Path.relative(
+                        FileSystem.instance.top_level_dir,
+                        Path.joinAbsString(
+                            FileSystem.instance.top_level_dir,
+                            &[_]string{
+                                source.path.name.dir,
+                                dependency_version.value.workspace.slice(buf),
+                            },
+                            .posix,
+                        ),
+                    ),
+                );
+                dependency_version.value.workspace = path;
```

Comment:
i wonder if it should be case insensitive

Reply:
its fine to not worry about it right now

Reply:
I think most file systems are case sensitive? :thinking: